cron.php

```diff
@@ -0,0 +1,464 @@
+<?php
+/*
+ Plugin Name: WP-Cron Control Revisited
+ Plugin URI:
+ Description: Take control of wp-cron execution.
+ Author: Erick Hitter, Automattic
+ Version: 0.1
+ Text Domain: wp-cron-control-revisited
+ */
+
+// Enable in production only for specific sites
+// Otherwise, class WP-Cron Control is loaded
+$whitelisted_sites = array();
+if ( VIP_GO_ENV && ! in_array( FILES_CLIENT_SITE_ID, $whitelisted_sites ) ) {
+	return;
+}
+
+class WP_Cron_Control_Revisited {
+	/**
+	 * Class instance
+	 */
+	private static $__instance = null;
+
+	public static function instance() {
+		if ( ! is_a( self::$__instance, __CLASS__ ) ) {
+			self::$__instance = new self;
+		}
+
+		return self::$__instance;
+	}
+
+	/**
+	 * PLUGIN SETUP
+	 */
+
+	/**
+	 * Class properties
+	 */
+	private $namespace = 'wp-cron-control-revisited/v1';
+	private $secret    = null;
+
+	private $job_queue_size                  = 10;
+	private $job_queue_window_in_seconds     = 60;
+	private $job_execution_buffer_in_seconds = 15;
+	private $job_timeout_in_minutes          = 10;
+	private $job_concurrency_limit           = 10;
```

Comment:
Initially, I'd thought about making these filterable, or something that could be controlled via the API requests, but at this point, I'm not sure there's a value in doing so. I'm going to leave this for now in case testing shows that we need to vary these settings based on site size/traffic/whatever the 🎱  might show.